src/goto-harness/function_harness_generator.h

```diff
@@ -0,0 +1,38 @@
+/******************************************************************\
+
+Module: harness generator for functions
+
+Author: Diffblue Ltd.
+
+\******************************************************************/
+
+#ifndef CPROVER_GOTO_HARNESS_FUNCTION_HARNESS_GENERATOR_H
+#define CPROVER_GOTO_HARNESS_FUNCTION_HARNESS_GENERATOR_H
+
+#include <memory>
+#include <string>
+
+#include <util/ui_message.h>
+
+#include "goto_harness_generator.h"
+
+/// Function harness generator that for a specified goto-function
+/// generates a harness that sets up its arguments and calls it.
+class function_harness_generatort : public goto_harness_generatort
+{
+public:
+  explicit function_harness_generatort(ui_message_handlert &message_handler);
+  ~function_harness_generatort() override;
```

Comment:
This one is required because we use the pimpl idiom here; `default_delete` requires its type parameter to be a complete type, so we can't instantiate the default destructor of `unique_ptr` here; Instead we do this in the makefile, where the full `implt` definition is visible.